libraries/Microsoft.Bot.Builder.Dialogs.Adaptive/Actions/BeginDialog.cs

```diff
@@ -51,16 +45,18 @@ public string Disabled
         /// The property path to store the dialog result in.
         /// </value>
         [JsonProperty("resultProperty")]
-        public string ResultProperty { get; set; }
+        public StringExpression ResultProperty { get; set; }
 
         public override async Task<DialogTurnResult> BeginDialogAsync(DialogContext dc, object options = null, CancellationToken cancellationToken = default(CancellationToken))
         {
             if (options is CancellationToken)
             {
                 throw new ArgumentException($"{nameof(options)} cannot be a cancellation token");
             }
+            
+            var dcState = dc.GetState();
 
-            if (this.disabled != null && (bool?)this.disabled.TryEvaluate(dc.GetState()).value == true)
+            if (this.Disabled != null && this.Disabled.GetValue(dcState) == true)
```

Comment:
>this.Disabled != null [](start = 16, length = 21)

This is testing to see if there is an expression assigned, right?

Reply:
yes

---
In reply to: [367266589](https://github.com/microsoft/botbuilder-dotnet/pull/3253#discussion_r367266589) [](ancestors = 367266589)